lib/s3.go

```diff
@@ -45,14 +45,14 @@ func (f *s3File) ETag() string {
 }
 
 func (f *s3File) Size() int64 {
-	return f.o.Size
+	return *f.o.Size
 }
 
-func newRemoteStore(cfg *Config, logger printer) (*s3Store, error) {
+func newRemoteStore(ctx context.Context, cfg *Config, logger printer) (*s3Store, error) {
```

Comment:
What is the ctx needed for here? It seems it's always contex.Backgroun.d

Reply:
It's a good question, do we want to create a background context for the default config factory or tie it to the deployer's context? I figured the latter but happy to update if you'd prefer.